tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py

```diff
@@ -219,3 +294,45 @@ def test_extra_valid_files(clean_files):
         host_manager.run_command(test_hosts[0], f'rm -rf {os.path.join(WAZUH_PATH, "queue", "agent-groups", id_)}')
     # Wait until they are deleted in all nodes.
     time.sleep(time_to_sync)
+
+
+@pytest.mark.parametrize('n_small_files', [5])
+def test_zip_size_limit(n_small_files, clean_files, update_cluster_json):
+    """Check if zip size limit works and if it is dynamically adapted.
+
+    Create several large files on the master. The time needed to send them all together
+    should exceed the maximum time allowed and a timeout should be generated on the workers.
+    The test verifies that:
+        - Workers notify the master to cancel the file being sent.
+        - The master reduces the zip size limit.
+        - The master warns that a file that is too large will not be synchronized.
+        - The master warns that not all the files could be synchronized since they did not fit in the zip.
+        - Eventually the zip size limit is increased again.
+        - The workers end up receiving all the files that do not exceed the maximum size.
+
+    Args:
+        n_small_files (int): Number of files that should be able to synchronize.
+    """
+    small_size = configuration['min_zip_size'] - 1024
+    big_size = configuration['max_zip_size'] + 1024
+
+    # Create a tmp folder and all files inside in the master node.
+    host_manager.run_command(test_hosts[0], f'mkdir {tmp_size_test_path}')
+    create_file(test_hosts[0], os.path.join(tmp_size_test_path, big_file_name), big_size)
+    for i in range(n_small_files):
+        create_file(test_hosts[0], os.path.join(tmp_size_test_path, file_prefix + str(i)), small_size)
+
+    # Move files from tmp folder to destination folder. This way, all files are synced at the same time.
+    host_manager.run_shell(test_hosts[0], f'mv {os.path.join(tmp_size_test_path, "test_*")} {dst_size_test_path}')
+
+    # Check whether zip size limit changed, big files are rejected, etc.
+    HostMonitor(inventory_path=inventory_path, messages_path=messages_path, tmp_path=tmp_path).run()
+
+    for host in worker_hosts:
+        # Make sure that smaller files were synced.
+        ls_count = host_manager.run_shell(host, f'ls {os.path.join(dst_size_test_path, file_prefix + "*")} | wc -l')
+        assert int(ls_count) == n_small_files, f'Less files in {host} ({ls_count}) than expected ({n_small_files}).'
+
+        # While too big files were rejected.
+        ls_big = host_manager.run_shell(host, f'ls {os.path.join(dst_size_test_path, big_file_name)}')
+        assert ls_big == '', f"File {big_file_name}' was expected not to be copied in {host}, but it was."
```

Comment:
```suggestion
    for host in worker_hosts:
        # Make sure that smaller files were synced.
        ls_count = host_manager.run_shell(host, f"ls {os.path.join(dst_size_test_path, file_prefix + '*')} | wc -l")
        assert int(ls_count) == n_small_files, f"Less files in {host} ({ls_count}) than expected ({n_small_files})."

        # While too big files were rejected.
        ls_big = host_manager.run_shell(host, f"ls {os.path.join(dst_size_test_path, big_file_name)}")
        assert ls_big == '', f"File '{big_file_name}' was expected not to be copied in {host}, but it was."
```

Also, we need to check that the synchronized files are the ones expected, not just check the number of files.